connector/src/main/java/org/jboss/as/connector/subsystems/datasources/EnableAttributeWriteHandler.java

```diff
@@ -0,0 +1,116 @@
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.connector.subsystems.datasources;
+
+import static org.jboss.as.connector.subsystems.datasources.Constants.ENABLED;
+import static org.jboss.as.connector.subsystems.datasources.Constants.JNDI_NAME;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OP_ADDR;
+
+import org.jboss.as.controller.AbstractWriteAttributeHandler;
+import org.jboss.as.controller.AttributeDefinition;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.OperationStepHandler;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.registry.ManagementResourceRegistration;
+import org.jboss.as.controller.registry.Resource;
+import org.jboss.dmr.ModelNode;
+import org.jboss.msc.service.ServiceRegistry;
+
+
+public class EnableAttributeWriteHandler extends AbstractWriteAttributeHandler<Void> {
+
+    private final boolean xa;
+
+    public EnableAttributeWriteHandler(final Boolean xa, final AttributeDefinition... definitions) {
+        super(definitions);
+        this.xa = xa;
+    }
+
+    @Override
+    public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+        super.execute(context, operation);
+
+        final ManagementResourceRegistration datasourceRegistration = context.getResourceRegistrationForUpdate();
+        final Resource resource = context.readResourceForUpdate(PathAddress.EMPTY_ADDRESS);
+        final ModelNode model = resource.getModel();
+
+        if (ENABLED.resolveModelAttribute(context,model).asBoolean()) {
+            if (context.isNormalServer()) {
+                DataSourceStatisticsListener.registerStatisticsResources(resource);
+
+                context.addStep(new OperationStepHandler() {
+                    public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+                        final ManagementResourceRegistration registration = context.getResourceRegistrationForUpdate();
+                        DataSourceEnable.addServices(context, operation, registration, model, xa);
+                        context.completeStep(new OperationContext.RollbackHandler() {
```

Comment:
If you don't need to take any action on rollback you used to call stepCompleted() instead. That is however no longer necessary after some changes Brian made a while ago